textdb/textdb-dataflow/src/main/java/edu/uci/ics/textdb/dataflow/regexmatch/RegexInfo.java

```diff
@@ -78,5 +78,100 @@ static RegexInfo anyChar() {
 		regexInfo.emptyable = false;
 		return regexInfo;
 	}
+	
+	/**
+	 * This function simplifies the regexInfo. <br>
+	 * If the "exact" set gets too large, it will add exact to the query tree,
+	 * and move relevant pieces into prefix and suffix. <br>
+	 * 
+	 * Simplification is performed under three circumstances. <br>
+	 * 1, the size ofexact set is larger than a threshold. <br>
+	 * 2, the minimum length of strings in exact is larger than {@code gramLength+1}, 
```

Comment:
Per our discussion, please add examples to the comments to make the logic easier to follow.